UsersApp/Controllers/LoginViewController.swift

```diff
@@ -0,0 +1,187 @@
+        if value.count > EmailRules.maxLength {return "Email must be at most \(EmailRules.maxLength) characters"}
+        let emailRegex = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+        let emailPredicate = NSPredicate(format: "SELF MATCHES %@", emailRegex)
+        if !emailPredicate.evaluate(with: value) {
+            return "Invalid Email Address format"
+        }
+        
+        return nil
+    }
+    
+    private func invalidPasswordFormat(_ value: String) -> String? {
+        if value.count == 0 {return "Required"}
+        if value.count < PasswordRules.minLength {return "Password must be at least \(PasswordRules.minLength) characters"}
+        if value.count > PasswordRules.maxLength {return "Password must be at most  \(PasswordRules.maxLength) characters"}
+        if value.contains(" ") {return "Password cannot contain spaces"}
+        if !containsDigits(value) {
+            return "Password must contain at least 1 digit"
+        }
+        
+        return nil
+    }
+    
+    private func containsDigits(_ value: String) -> Bool {
+        let passwordRegex = ".*[0-9]+.*"
+        let passwordPredicate = NSPredicate(format: "SELF MATCHES %@", passwordRegex)
+        return passwordPredicate.evaluate(with: value)
+    }
+    
+    private func checkForValidForm() {
+        if emailErrorTF.isHidden && passwordErrorTF.isHidden {
+            loginButton.isEnabled = true
+        } else {
+            loginButton.isEnabled = false
+        }
+    }
+    
+    private func makeUsersViewController() {
+        let usersViewController = UsersViewController()
+        
+        let navController = UINavigationController(rootViewController: usersViewController)
+        navController.modalPresentationStyle = .fullScreen
+
+        present(navController, animated: true, completion: nil)
+    }
+}
+
+// MARK: - Keyboard Behaviour
+extension LoginViewController {
+    private func handleKeyboardBehaviour() {
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
+        view.addGestureRecognizer(tapGesture)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification: )), name: UIResponder.keyboardWillShowNotification, object: nil)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification: )), name: UIResponder.keyboardWillHideNotification, object: nil)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification: )), name: UIResponder.keyboardWillChangeFrameNotification, object: self)
+    }
+    
+    @objc private func keyboardWillShow(notification: NSNotification) {
+        if notification.name == UIResponder.keyboardWillShowNotification ||
+        notification.name == UIResponder.keyboardWillChangeFrameNotification {
+            view.frame.origin.y = -40
```

Comment:
do not force with hardcoded frae values. 
you should use dynamic mesures based on Keyboard height which can differ is you have suggestions or not, etc